### PR TITLE
887 validate required fields for internal apps

### PR DESCRIPTION
**Pull request by mmaquina**
## Description

Add validation when saving time entries of internal projects. They must have a description or a ticket name(uri).

## Files changed

- src/app/modules/shared/components/details-fields/details-fields.component.spec.ts
- src/app/modules/shared/components/details-fields/details-fields.component.ts
- src/app/modules/time-clock/components/entry-fields/entry-fields.component.spec.ts
- src/app/modules/time-clock/components/entry-fields/entry-fields.component.ts
- src/app/modules/shared/messages.ts
- src/app/modules/shared/internal-app-constants.ts

## Task board

- [Internal Apps required](https://ioetec.atlassian.net/browse/TTL-887)

### CODE REVIEW

File: src/app/modules/shared/components/details-fields/details-fields.component.ts [hunk 1]
```diff
@@ -331,6 +331,14 @@ export class DetailsFieldsComponent implements OnChanges, OnInit {
   }
 
   onSubmit() {
+
+    if (this.entryForm.value.project_name.includes('ioet')) {
+      if (this.entryForm.value.uri === '' && this.entryForm.value.description === '') {
```
[the remark this answers is not in the document]
Comment (thegreatyamori) [reply src/app/modules/shared/components/details-fields/details-fields.component.ts:336]: see more info: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every

File: src/app/modules/time-clock/components/entry-fields/entry-fields.component.ts [hunk 2]
```diff
@@ -192,4 +198,15 @@ export class EntryFieldsComponent implements OnInit, OnDestroy {
     this.loadActiveEntrySubscription.unsubscribe();
     this.actionSetDateSubscription.unsubscribe();
   }
+
+  requiredFieldsForInternalAppExist(customerName) {
+    const emptyFields = this.entryForm.value.uri === '' && this.entryForm.value.description === '';
+    const isInternalApp = customerName.includes('ioet');
```
Comment (thegreatyamori) [inline src/app/modules/time-clock/components/entry-fields/entry-fields.component.ts:204]: is this rule skipping those projects? or are need to skip those projects?
- ioet-internal
- other trainning
- safari books
- english lessons
- university

**Review by thegreatyamori — CHANGES_REQUESTED** [review]

File: src/app/modules/time-clock/components/entry-fields/entry-fields.component.ts (hunk 2 above)
Comment (mmaquina) [reply src/app/modules/time-clock/components/entry-fields/entry-fields.component.ts:204]: we are not skipping those projects, as it is, they should add a description

Comment (mmaquina) [reply src/app/modules/time-clock/components/entry-fields/entry-fields.component.ts:204]: which I think is ok, users should describe.... except from english lessons, there is not really much to describe there

**Review by thegreatyamori — APPROVED** [review]

**Review by thegreatyamori — APPROVED** [review]

**Pull request merged**